pkgs/tools/networking/firefoxpwa/default.nix

```diff
@@ -0,0 +1,87 @@
+{ stdenv
+, rustPlatform
+, fetchFromGitHub
+, openssl
+, symlinkJoin
+, buildFHSUserEnv
+, pkg-config
+, installShellFiles
+, lib
+}:
+let
+  version = "2.4.1";
+  dir = "native";
+  source = fetchFromGitHub {
+    owner = "filips123";
+    repo = "PWAsForFirefox";
+    rev = "v${version}";
+    sha256 = "sha256-AkCTfbWsAF6N1hmowK33XzI2bnBlbzML9IiGjujc8N8=";
+    sparseCheckout = [ dir ];
+  };
+  pname = "firefoxpwa";
+
+  unwrapped = rustPlatform.buildRustPackage {
+    inherit version;
+    pname = "${pname}-unwrapped";
+
+    src = "${source}/${dir}";
+    cargoSha256 = "sha256-BHEFBrDrSZv9Bwl4q+MQA1BAoaJOxsNc/IK6ygrjwVY=";
+
+    doCheck = false;
+
+    nativeBuildInputs = [ pkg-config installShellFiles ];
+    buildInputs = [ openssl.dev openssl ];
+
+    preConfigure = ''
```

Comment:
Can we do this in postPatch?

Reply:
No or cargo complains

Reply:
That means that our cargo.lock is out of date and needs to be fixed, too.

Reply:
It's because I don't want to update the `version` in `Cargo.toml` every time before I release a new version. So, `Cargo.toml` in the repository only sets `version = "0.0.0"` and the actual version is then automatically set when building and releasing the package (in GitHub Actions for the official packages and here for Nix package).

Reply:
We should patch that version to be correct.

Reply:
Aren't we already doing this?